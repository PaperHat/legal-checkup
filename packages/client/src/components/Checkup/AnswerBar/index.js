import React from 'react'

import { NO, NOT_SURE, YES } from '../../../state/constants'

import Bar from './Bar'
import ResponseButton from './ResponseButton'

export default function AnswerBar () {
  return (
    <Bar>
      <ResponseButton type={YES} />
      <ResponseButton type={NO} />
      <ResponseButton type={NOT_SURE} />
    </Bar>
  )
}
