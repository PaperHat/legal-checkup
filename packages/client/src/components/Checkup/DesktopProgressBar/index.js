import React from 'react'

// import makeQuestionButton from '../../wrappers/makeQuestionButton'
import makeNextQuestionButton from '../../wrappers/makeNextQuestionButton'
import makePreviousQuestionButton
  from '../../wrappers/makePreviousQuestionButton'

import NextButton from './NextButton'
import PreviousButton from './PreviousButton'
import ProgressBar from './ProgressBar'

// import QuestionButton from './QuestionButton'

// const QuestionBtn = makeQuestionButton(QuestionButton)
const ConnectedNextButton = makeNextQuestionButton(NextButton)
const ConnectedPreviousButton = makePreviousQuestionButton(PreviousButton)

export default function DesktopProgressBar () {
  return (
    <ProgressBar>
      <ConnectedPreviousButton />
      {/* map through question numbers and add QuestionButton components here */}
      <ConnectedNextButton />
    </ProgressBar>
  )
}
