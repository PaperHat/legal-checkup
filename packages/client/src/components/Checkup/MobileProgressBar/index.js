import React from 'react'

import makeNextQuestionButton from '../../../wrappers/makeNextQuestionButton'
import makePreviousQuestionButton
  from '../../../wrappers/makePreviousQuestionButton'

import CurrentTopic from './CurrentTopic'
import NextButton from './NextButton'
import PreviousButton from './PreviousButton'
import ProgressBar from './ProgressBar'

const ConnectedNextButton = makeNextQuestionButton(NextButton)
const ConnectedPreviousButton = makePreviousQuestionButton(PreviousButton)

export default function MobileProgressBar () {
  return (
    <ProgressBar>
      <ConnectedPreviousButton />
      <CurrentTopic />
      <ConnectedNextButton />
    </ProgressBar>
  )
}
