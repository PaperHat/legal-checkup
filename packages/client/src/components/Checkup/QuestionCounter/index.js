import { connect } from 'react-redux'

import {
  getActiveQuestionIndex,
  getQuestionCount
} from '../../../state/selectors'

import Counter from './Counter'

function mapStateToProps ({ checkup }) {
  const children = `${getActiveQuestionIndex(checkup) + 1} / ${getQuestionCount(
    checkup
  )}`

  return { children }
}

export default connect(mapStateToProps)(Counter)
