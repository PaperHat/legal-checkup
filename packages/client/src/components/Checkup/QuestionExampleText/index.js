import { connect } from 'react-redux'

import { getActiveQuestion } from '../../../state/selectors'
import StyledQuestionExampleText from './ExampleText'

function mapStateToProps ({ checkup }) {
  const { help: children } = getActiveQuestion(checkup)

  return {
    children
  }
}

export default connect(mapStateToProps)(StyledQuestionExampleText)
