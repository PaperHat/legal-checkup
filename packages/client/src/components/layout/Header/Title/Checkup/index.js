import styled from 'styled-components'

import { MOBILE, WHITE, RED_ACCENT } from '../../../../../constants'

const Checkup = styled.div`
  color: ${({ format }) =>
    format === MOBILE ? `#${WHITE}` : `#${RED_ACCENT}`};
  line-height: 1;
  text-transform: uppercase;
  font-size: inherit;
`
Checkup.displayName = 'Checkup'

export default Checkup
