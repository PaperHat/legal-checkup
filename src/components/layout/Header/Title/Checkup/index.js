import styled from 'styled-components'

import { MOBILE, WHITE, TITLE_TEXT } from '../../../../../constants'

const Checkup = styled.div`
  color: ${({ format }) => (format === MOBILE ? `${WHITE}` : `${TITLE_TEXT}`)};
  line-height: 1;
  text-transform: uppercase;
  font-size: inherit;
`
Checkup.displayName = 'StyledCheckup'

export default Checkup
