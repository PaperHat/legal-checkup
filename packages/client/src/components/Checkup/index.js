import * as React from 'react'
import { Fragment } from 'react'

import AnswerBar from './AnswerBar'
import MobileProgressBar from './MobileProgressBar'
import QuestionBox from './QuestionBox'
import QuestionCounter from './QuestionCounter'

export default function Checkup () {
  return (
    <Fragment>
      <MobileProgressBar />
      <QuestionCounter />
      <QuestionBox />
      <AnswerBar />
    </Fragment>
  )
}
