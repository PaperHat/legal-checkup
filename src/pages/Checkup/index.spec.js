import React from 'react'
import { shallow } from 'enzyme'

import Checkup from '.'

describe('components:App', () => {
  it('renders the App with a blank game board and nine squares', () => {
    expect(toJson(shallow(<Checkup />).dive())).toMatchSnapshot()
  })
})
