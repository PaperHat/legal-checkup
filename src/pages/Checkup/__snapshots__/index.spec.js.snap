// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:Checkup renders the Checkup page 1`] = `
<Layout>
  <QuestionBox />
</Layout>
`;
